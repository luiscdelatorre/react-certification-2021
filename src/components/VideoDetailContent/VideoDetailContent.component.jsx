import React from 'react';
import Styled from './VideoDetailContent.styles';

const VideoDetailContent = ({ channel, channelLogo, subscriberCount, description }) => {
  return (
    <div>
      <Styled.DescriptionContainer>
        <Styled.DescriptionLogo src={channelLogo} alt="channel" />
        <hgroup>
          <Styled.Title>{channel}</Styled.Title>
          <Styled.Subtitle>{subscriberCount} Subscribers</Styled.Subtitle>
        </hgroup>
      </Styled.DescriptionContainer>
      <Styled.Description>{description}</Styled.Description>
    </div>
  );
};

export default VideoDetailContent;
