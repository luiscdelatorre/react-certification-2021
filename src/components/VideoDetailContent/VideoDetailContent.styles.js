import styled from 'styled-components';
import device from '../../config/device';

const DescriptionContainer = styled.div`
  display: flex;
  align-items: center;
`;

const DescriptionLogo = styled.img`
  border-radius: 50%;
  width: 4rem;
  height: 4rem;
  margin-right: 1rem;
`;

const Title = styled.h5`
  color: ${(props) => props.theme.cardTitle};
`;

const Subtitle = styled.h6`
  color: ${(props) => props.theme.textSecondary};
`;

const Description = styled.p`
  margin-top: 1rem;
  color: ${(props) => props.theme.cardSubtitle};
  @media ${device.laptop} {
    padding-left: 5rem;
  }
`;

const Styled = {
  DescriptionContainer,
  DescriptionLogo,
  Description,
  Title,
  Subtitle,
};

export default Styled;
