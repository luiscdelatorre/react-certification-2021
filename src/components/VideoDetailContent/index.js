export { default } from './VideoDetailContent.component';
