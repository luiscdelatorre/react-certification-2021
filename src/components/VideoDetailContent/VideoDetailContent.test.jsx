import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import detail from '../../mock/youtube-detail-mock.json';
import channel from '../../mock/youtube-channel-mock.json';
import VideoDetailContent from './VideoDetailContent.component';

describe('Video Detail Content component', () => {
  it('Should display mock data in component', () => {
    const video = detail.items[0];
    const videoChannel = channel.items[0];
    render(
      <VideoDetailContent
        channel={videoChannel.snippet.title}
        channelLogo={videoChannel.snippet.thumbnails.default.url}
        subscriberCount={videoChannel.statistics.subscriberCount}
        description={video.snippet.description}
      />
    );

    expect(screen.queryByText(videoChannel.snippet.title).tagName).toBe('H5');
    expect(
      screen.queryByText(`${videoChannel.statistics.subscriberCount} Subscribers`).tagName
    ).toBe('H6');
  });

  it('Validates Video Detail snapshot', () => {
    const videoChannel = channel.items[0];
    const video = detail.items[0];
    const component = renderer.create(
      <VideoDetailContent
        channel={videoChannel.snippet.title}
        channelLogo={videoChannel.snippet.thumbnails.default.url}
        subscriberCount={video.statistics.subscriberCount}
        description={video.snippet.description}
      />
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
