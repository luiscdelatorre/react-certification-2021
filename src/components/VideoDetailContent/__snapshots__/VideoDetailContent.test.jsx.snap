// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video Detail Content component Validates Video Detail snapshot 1`] = `
<div>
  <div
    className="sc-bdfBwQ crBVTC"
  >
    <img
      alt="channel"
      className="sc-gsTCUz jdaUzS"
      src="https://yt3.ggpht.com/ytc/AAUvwnighSReQlmHl_S_vSfvnWBAG5Cw4A0YxtE0tm5OpQ=s88-c-k-c0x00ffffff-no-rj"
    />
    <hgroup>
      <h5
        className="sc-dlfnbm edLazA"
      >
        Wizeline
      </h5>
      <h6
        className="sc-hKgILt czLCvW"
      >
         Subscribers
      </h6>
    </hgroup>
  </div>
  <p
    className="sc-eCssSg cTpCtJ"
  >
    Official music video by The Weeknd performing "Save Your Tears"– 'After Hours' available everywhere now: http://theweeknd.co/afterhoursYD
 
►Subscribe to The Weeknd on YouTube: http://theweeknd.co/subscribeYD 
 
►Get exclusive merch: https://shop.theweeknd.com/
 
►Follow The Weeknd:
https://twitter.com/theweeknd 
https://www.facebook.com/theweeknd 
https://www.instagram.com/theweeknd 
https://www.theweeknd.com 

Directed by Cliqua
Produced by Roisín Audrey Moloney
Executive Producer: Jerad Anderson
Production Company: Florence
Abel’s Costume Designer: Matt Henson
Prosthetics: Mike Marino, Prosthetic Renaissance
Director of Photography: Xiaolong Liu
Production Designer: Annie Sperling
1st AD: Mike Alberts
Production Supervisor: Tori Storosh
Production Supervisor: Joe Keenan
Assist. Production Supervisor: Linda Nhem
Cast Costume Designer: Lisa Madonna
Gaffer: Matt Ardine
Key Grip: Jon Booker
Camera Operator Nick Muller
Steadicam Colin MacDonnell
Editor: Miles Trahan
Color: Matt Osbourne @ Company 3
Sound Design: Christian Stropko
VFX: Buralqy + MADNOMAD
Online: Sunset Edit
Titles: Bradley Pinkerton

Lyrics:

Ooh
Na na, yeah
I saw you dancing in a crowded room
You look so happy when I'm not with you
But then you saw me, caught you by surprise
A single teardrop falling from your eye
I don't know why I run away
I make you cry when I run away
You could've asked me why I broke your heart
You could've told me that you fell apart
But you walked past me like I wasn't there
And just pretended like you didn't care
I don't know why I run away
I make you cry when I run away
Take me back 'cause I wanna stay
Save your tears for another
Save your tears for another day
Save your tears for another day
So
I made you think that I would always stay
I said some things that I should never say
Yeah, I broke your heart like someone did to mine
And now you won't love me for a second time
I don't know why I run away, oh, girl
Said I make you cry when I run away
Girl, take me back 'cause I wanna stay
Save your tears for another
I realize that I'm much too late
And you deserve someone better
Save your tears for another day (Ooh, yeah)
Save your tears for another day (Yeah)
I don't know why I run away
I make you cry when I run away
Save your tears for another day, ooh, girl (Ah)
I said save your tears for another day (Ah)
Save your tears for another day (Ah)
Save your tears for another day (Ah)

#TheWeeknd #SaveYourTears #AfterHours

Music video by The Weeknd performing Save Your Tears. © 2021 The Weeknd XO, Inc., manufactured and marketed by Republic Records, a division of UMG Recordings, Inc.

http://vevo.ly/7MOdnq
  </p>
</div>
`;
